### PR TITLE
Clean up duckhunt.py and refactor game state logic

### PR DESCRIPTION
**Pull request by linuxdaemon**

### CODE REVIEW

File: plugins/duckhunt.py [hunk 1]
```diff
@@ -170,44 +223,58 @@ def start_hunt(db, chan, message, conn):
...
+    """
+    status = get_state_table(conn, chan)  # type: ChannelState
+    status.next_duck_time = random.randint(int(time()) + 480, int(time()) + 3600)
+    # status.flyaway = status.next_duck_time + 600
+    status.duck_status = 0
     # let's also reset the number of messages said and the list of masks that have spoken.
-    game_status[conn][chan]['messages'] = 0
-    game_status[conn][chan]['masks'] = []
+    status.messages = 0
+    status.masks = []
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py]: Any reason to not use `status.masks.clear()` here?

File: plugins/duckhunt.py [hunk 2]
```diff
@@ -225,43 +292,53 @@ def generate_duck():
...
+    :type nick: str
+    :type chan: str
+    :type db: sqlalchemy.orm.Session
+    :type conn: cloudbot.client.Client
+    :type shoot: int
+    :type friend: int
+    """
     query = table.insert().values(
         network=conn.name,
         chan=chan.lower(),
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py:355]: As this is a cleanup, please fix the ending paren on this call and add a newline after it

File: plugins/duckhunt.py [hunk 3]
```diff
@@ -273,7 +350,14 @@ def dbadd_entry(nick, chan, db, conn, shoot, friend):
...
+    :type nick: str
+    :type chan: str
+    :type db: sqlalchemy.orm.Session
+    :type conn: cloudbot.client.Client
+    :type shoot: int
+    :type friend: int
+    """
     if shoot and not friend:
         query = table.update() \
             .where(table.c.network == conn.name) \
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py]: newline after this call for readability, same with the ones below.

File: plugins/duckhunt.py [hunk 4]
```diff
@@ -652,7 +814,12 @@ def ducks_user(text, nick, chan, conn, db, message):
 
 @hook.command("duckstats", autohelp=False)
 def duck_stats(chan, conn, db, message):
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py:842]: Break up the logical blocks in this function with newlines

File: plugins/duckhunt.py [hunk 5]
```diff
@@ -302,28 +386,45 @@ def dbupdate(nick, chan, db, conn, shoot, friend):
...
+    :type db: sqlalchemy.orm.Session
+    :type conn: cloudbot.client.Client
+    :type shoot: int
+    :type friend: int
+    """
     score = db.execute(select([table.c.shot, table.c.befriend])
                        .where(table.c.network == conn.name)
                        .where(table.c.chan == chan.lower())
                        .where(table.c.name == nick.lower())).fetchone()
     if score:
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py:408]: add a newline for readability

File: plugins/duckhunt.py [hunk 6]
```diff
@@ -127,41 +173,48 @@ def save_status(db, _sleep=True):
...
+    :type event: cloudbot.event.Event
+    :type conn: cloudbot.client.Client
+    """
+    if is_opt_out(conn.name, event.chan):
         return
-    if game_status[conn.name][event.chan]['game_on'] == 1 and game_status[conn.name][event.chan]['duck_status'] == 0:
-        game_status[conn.name][event.chan]['messages'] += 1
-        if event.host not in game_status[conn.name][event.chan]['masks']:
-            game_status[conn.name][event.chan]['masks'].append(event.host)
+    get_state_table(conn.name, event.chan).handle_message(event)
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py:206]: Add a newline here

**Review by A-UNDERSCORE-D — CHANGES_REQUESTED** [review]

File: plugins/duckhunt.py (hunk 1 above)
Comment (linuxdaemon) [reply plugins/duckhunt.py]: Because originally the field was only initialized there when it was still the dict structure. I'll change it.

**Review by A-UNDERSCORE-D — APPROVED** [review]

**Pull request merged**